### ISSUE TITLE
LiveOS ISOs cannot boot when SELinux is enabled

### ISSUE THREAD

**Issue by gmileka** [dracut-ng/dracut#1042]
**Describe the bug**
For a LiveOS ISO, when SELinux is enabled on the kernel command line and/or the rootfs SELinux configuration, the ISO does not boot.
Our LiveOS ISO initrd contains:
- the dmsquash-live module
- the overlayfs module
- the selinux-policy module
(and all their dependencies)

Our Investigation:

When we turned on permissive mode to see the failures, they can be split into two groups:
1. all the systemd daemons that start during initrd are failing to access file descriptor marked with kernel_t.
2. the overlay folders (/run/overlayfs, /run/ovlwork) and the /sysroot folders are marked tmpfs_t while they need to be root_t.

We have a potential fix here: https://github.com/microsoft/azurelinux/pull/11198
The fix updates the policy to cover the denials (1 above) and adds an overlay pre-pivot script to fix the `root_t` type. I opted to not include the `root_t` in the policy to limit it to when the overlayfs module is actually doing something.

I am not sure if this selinux/dmsquash-live/overlayfs combination has a better/offical way of configuring - or what I'm doing is basically the right thing.

**Distribution used**
Which distribution was this behaviour seen in?
Azure Linux 3.0

**Dracut version**
Which dracut version was this behaviour seen in?
102

**Init system**
Which init system is being used?
systemd

**To Reproduce**
Steps or code to reproduce the behavior.
- Create an Azure Linux LiveOS ISO image where;
  - SELinux is enabled (permissive).
- Log-in
- journalctl | grep -i avc

**Expected behavior**
The LiveOS dracut modules work smoothly when SELinux is enabled.

**Additional context**
Add any other context you like about the problem here.
I'm basically trying to understand if there is an official way of supporting the combination and I missed something, and if not - whether what I am doing makes sense or there is a better way.

**Comment by Conan-Kudo** [comment]
This is an odd issue, as Fedora live media all have SELinux enabled. Is the rootfs fully labeled inside the ISO image?

**Comment by gmileka** [comment]
@Conan-Kudo , yes, the rootfs is fully labeled.
Does anybody know by any chance whether live media has been working on Fedora since Dracut 102? I wonder if newer versions of Dracut fix the issue...

**Comment by Conan-Kudo** [comment]
Yes, Fedora Rawhide images are on Dracut 103 now. You can find them at https://nightly.fedoraproject.org/.

**Comment by FGrose** [comment]
@meeuw If `chcon -t root_t / /run/overlayfs /run/ovltest` is added to `/usr/libexec/livesys/livesys-main`, which runs in Fedora soon after startup, the labels are set as desired.  Is this remedy sufficient for your cases?

**Comment by devkontrol** [comment]
CC @christopherco for Azure

**Comment by iavael** [comment]
This issue also occurs in installed Fedora Silverblue with transient root, not just in LiveOS
